src/v/cluster/types.cc

```diff
@@ -957,11 +982,11 @@ adl<cluster::incremental_topic_updates>::from(iobuf_parser& in) {
       = adl<cluster::property_update<tristate<std::chrono::milliseconds>>>{}
           .from(in);
 
-    if (version < 0) {
-        updates.data_policy
-          = adl<
-              cluster::property_update<std::optional<v8_engine::data_policy>>>{}
-              .from(in);
+    if (
+      version == cluster::incremental_topic_updates::version_with_data_policy) {
+        // Unused data_policy property from log
```

Comment:
Would be good to extend comment to say this property was _never_ used, to clarify we're not dropping anything useful.